### PR TITLE
parabricks/applybqsr version updated and tests migrated to nf-test

### LINKED ISSUE #6861: update module: parabricks/applybqsr
### Is there an existing module for this?

- [x] I have searched for the existing module

### Is there an open PR for this?

- [x] I have searched for existing PRs

### Is there an open issue for this?

- [x] I have searched for existing issues

### Are you going to work on this?

- [ ] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

Also port to nf-test

### PR DESCRIPTION
**Pull request by gitanoqevaporelmundoentero**
Tools applybqsr is updated to latest version (4.3.2-1) and tests were migrated to new convention (from pytest to nf-test).

## PR checklist

Closes #6861 

- [X] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [X] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: .github/workflows/test.yml
Comment (famosab) [inline .github/workflows/test.yml:1]: I think you still need to remove it from the part of the file with pytest.

File: tests/modules/nf-core/parabricks/applybqsr/test_GPU_config.txt
Comment (famosab) [inline tests/modules/nf-core/parabricks/applybqsr/test_GPU_config.txt:1]: move this file to the test folder just to not loose the information - we might delete it later though

**Review by famosab — COMMENTED** [review]
Looks good so far.

**Comment by famosab** [comment]
We need to change the way the GPUs are accessed in the main.nf file. Maybe we can do that here aswell. 

so the line
`--num-gpus $task.accelerator.request \\`

needs to be
`$num_gpus \\`

and add this above the script block
`def num_gpus = task.accelerator ? "--num-gpus $task.accelerator.request" : ''`

File: modules/nf-core/parabricks/applybqsr/tests/test_GPU_config.txt
Comment (sateeshperi) [inline modules/nf-core/parabricks/applybqsr/tests/test_GPU_config.txt:1]: unused file ?

Comment (sateeshperi) [inline modules/nf-core/parabricks/applybqsr/tests/test_GPU_config.txt:1]: is this a config ?

Comment (famosab) [reply modules/nf-core/parabricks/applybqsr/tests/test_GPU_config.txt:1]: that was in the old tests folder from the pytest

Comment (sateeshperi) [reply modules/nf-core/parabricks/applybqsr/tests/test_GPU_config.txt:1]: oh not needed then once the gpu profile is set

**Comment by mashehu** [comment]
I think I finally got everything running, and I blame outdated snapshots for the failing tests. Am I wrong there, @gitanoqevaporelmundoentero, @famosab

**Comment by famosab** [comment]
@sateeshperi @sateeshperi (not sure which one is the correct handle)
For fq2bammeth there is also no distinction between docker & singularity. Do you know why?

@mashehu @gitanoqevaporelmundoentero I will try to update the snaps on our cluster and push them.

**Comment by sateeshperi** [comment]
honestly no idea since I didn't test using singularity. I just borrowed that config block from nf-core/scdownstream

**Comment by famosab** [comment]
> could you try what we had previously ?

No, I think the AWS instance was closed. I am trying to run it on other stuff we have locally but each one of those has some limitations.

**Comment by mashehu** [comment]
restarted the test instance

**Comment by famosab** [comment]
Snaps are updated. I could not solve the weird `fq2bammeth` behaviour. 
Seems like the test for applybqsr somehow does not pick up and the singularity one for fq2bam fails?

**Comment by mashehu** [comment]
singularity hiccup is fixed. the other tests shouldn't be picked up...

**Comment by famosab** [comment]
@sateeshperi Now the fq2bammeth test fails with
`.command.sh: line 2: bwameth.py: command not found`

any ideas?

**Comment by sateeshperi** [comment]
that means the container profile is not invoked properly....hmmmm

**Comment by famosab** [comment]
@sateeshperi if you are fine with that then give us the approval :D

**Comment by famosab** [comment]
> singularity hiccup is fixed. the other tests shouldn't be picked up...

I think the issue is back, back again @mashehu

**Comment by mashehu** [comment]
There is a weird ci run, where the profile is empty. No idea how that can be. Will investigate tomorrow

**Comment by famosab** [comment]
@mashehu ping :)

**Comment by famosab** [comment]
Now only docker `fq2bam` is failing with
```
     >   [PB Error 2024-Nov-15 11:59:36][src/main.cu:997] CUDA_CHECK() failed with no CUDA-capable device is detected (100), exiting.
    >   For technical support visit https://docs.nvidia.com/clara/index.html#parabricks
    >   Exiting...
```

**Comment by famosab** [comment]
If we can fix this and merge it all the other parabricks modules can be updated accordingly and a bunch of PRs closed :)

**Review by sateeshperi — APPROVED** [review]

**Pull request merged**